tests/functional/tests/bundles/base.yaml

```diff
@@ -1,16 +1,18 @@
 description: "charm-juju-local charm test bundle"
 machines:
   "0":
-    constraints: "arch=amd64"
-relations:
-  - [ "server:juju-info", "juju-local:juju-info" ]
+    constraints: arch=amd64
 applications:
   server:
     charm: ubuntu
     num_units: 1
+    constraints: virt-type=virtual-machine
```

Comment:
I was testing this manually why it's not working, and after while I realized that you are defining `constraints` in two places, in machines and here. Please move this configuration to line 4.